packages/vscode-boxel-tools/src/skills.ts

```diff
@@ -35,6 +82,148 @@ export class SkillsProvider
     return selectedSkills;
   }
 
+  updateCursorRules(): void {
+    // Get all selected skills and write to .cursorrules file
+    const selectedSkills = this.getSelectedSkills();
+    let combinedText = '';
+
+    for (const skill of selectedSkills) {
+      combinedText += `${skill.instructions}\n`;
+    }
+
+    // Get the workspace folder
+
+    console.log('Cursor rules path: ', this.cursorRulesPath);
+
+    try {
+      fs.writeFileSync(this.cursorRulesPath, combinedText);
+      console.log('.cursorrules file updated successfully');
+    } catch (error) {
+      console.error('Error writing .cursorrules file:', error);
+      vscode.window.showErrorMessage('Failed to write .cursorrules file');
+    }
+  }
+
+  // Save skills to storage
+  private saveSkillsToStorage(): void {
+    try {
+      const skillsDataPath = path.join(this.dataPath, 'skills_data.json');
+
+      // Convert skill lists to a format suitable for storage, preserving the hierarchy
+      const skillListsData = this.skillLists.map((list) => ({
+        label: list.label,
+        realmUrl: list.realmUrl,
+        readOnly: list.readOnly,
+        skills: list.skills.map((skill) => ({
+          id: skill.id,
+          label: skill.label,
+          instructions: skill.instructions,
+        })),
+      }));
+
+      const dataToStore: StoredSkillData = {
+        skillLists: skillListsData,
+        lastFetched: Date.now(),
+      };
+
+      fs.writeFileSync(skillsDataPath, JSON.stringify(dataToStore, null, 2));
+      console.log('Skills saved to storage successfully');
+    } catch (error) {
+      console.error('Error saving skills to storage:', error);
+    }
+  }
+
+  // Save skill state (enabled/disabled) separately
+  private saveSkillState(): void {
+    try {
+      const stateFilePath = path.join(this.dataPath, 'skill_state.json');
+
+      // Get all enabled skill IDs
+      const enabledIds = this.skillLists
+        .flatMap((skillList) => skillList.skills)
+        .filter(
+          (skill) =>
+            skill.checkboxState === vscode.TreeItemCheckboxState.Checked,
+        )
+        .map((skill) => skill.id);
+
+      const stateData: SkillStateData = {
+        enabledSkillIds: Array.from(enabledIds),
+        lastUpdated: Date.now(),
+      };
+
+      fs.writeFileSync(stateFilePath, JSON.stringify(stateData, null, 2));
+      console.log('Skill state saved successfully');
+    } catch (error) {
+      console.error('Error saving skill state:', error);
+    }
+  }
+
+  // Load skills from storage
+  private loadSkillsFromStorage(): void {
```

Comment:
Why store/read skill details to/from here rather than storing references to the skill instance json?

Reply:
Yeah I should change that to store the json as is